ResourceStatusSystem/Client/SiteStatus.py

```diff
@@ -326,25 +244,66 @@ def setSiteStatus( self, site, status, comment = 'No comment' ):
     # fix case sensitive string
     status = status.capitalize()
     allowedStateList = ['Active', 'Banned', 'Degraded', 'Probing', 'Error', 'Unknown']
+
     if status not in allowedStateList:
       return S_ERROR(errno.EINVAL, 'Not a valid status, parameter rejected')
 
-    result = getProxyInfo()
-    if result['OK']:
-      tokenOwner = result['Value']['username']
+
+    if self.rssFlag:
+      result = getProxyInfo()
+      if result['OK']:
+        tokenOwner = result['Value']['username']
+      else:
+        return S_ERROR( "Unable to get user proxy info %s " % result['Message'] )
+
+      tokenExpiration = datetime.utcnow() + timedelta( days = 1 )
+
+      self.rssCache.acquireLock()
+      result = self.rsClient.modifyStatusElement( 'Site', 'Status', status = status, name = site,
+                                                  tokenExpiration = tokenExpiration, reason = comment,
+                                                  tokenOwner = tokenOwner )
+      if result['OK']:
+        self.rssCache.refreshCache()
+      else:
+        _msg = 'Error updating status of site %s to %s' % ( site, status )
+        gLogger.warn( 'RSS: %s' % _msg )
+
+      # Release lock, no matter what.
+      self.rssCache.releaseLock()
+
     else:
-      return S_ERROR( "Unable to get user proxy info %s " % result['Message'] )
+      if status in ['Active', 'Degraded']:
+        result = RPCClient('WorkloadManagement/WMSAdministrator').allowSite()
+      else:
+        result = RPCClient('WorkloadManagement/WMSAdministrator').banSite()
+
+    return result
+
+
 
-    tokenExpiration = datetime.utcnow() + timedelta( days = 1 )
 
-    result = self.rsClient.modifyStatusElement( 'Site', 'Status', status = status, name = site,
-                                                tokenExpiration = tokenExpiration, reason = comment,
-                                                tokenOwner = tokenOwner )
+def getCacheDictFromRawData( rawList ): #FIXME: to remove?
+  """
+  Formats the raw data list, which we know it must have tuples of four elements.
+  ( element1, element2 ) into a dictionary of tuples with the format
+  { ( element1 ): element2 )}.
+  The resulting dictionary will be the new Cache.
+
+  It happens that element1 is elementName,
+                  element4 is status.
+
+  :Parameters:
+    **rawList** - `list`
+      list of three element tuples [( element1, element2 ),... ]
+
+  :return: dict of the form { ( elementName ) : status, ... }
+  """
 
-    if not result['OK']:
-      return result
+  res = {}
```

Comment:
`res = dict( (r[0],r[1]) for r in rawList)`
why did you put a parenthesis around entry[0]? If you need a tuple, put a comma...

Reply:
Yes, this is just a copy-paste. The comment at line 285 is "FIXME: to remove?" But not in this PR.